fecgov/fec-cms
# CMS website status 500 page

## Summary

- Resolves #3742 
_Introduces new 500 error page template under a feature flag. This alternative 500 error message can be turned on with a redeploy._

## Impacted areas of the application

List general components of the application that this PR will affect:

-  500 error page messaging

## Screenshots

### Normal/Default 500 error message page
<img width="1322" alt="Screen Shot 2020-05-12 at 11 38 53 AM" src="https://user-images.githubusercontent.com/12799132/81714720-5a7cbd00-9445-11ea-9bac-6d2ad00ae774.png">

### Alternative 500 error message page:
<img width="1318" alt="Screen Shot 2020-05-12 at 11 39 29 AM" src="https://user-images.githubusercontent.com/12799132/81714754-68324280-9445-11ea-99ae-0071d66870a8.png">

## How to test

- Pull down the branch
- Turn off DEBUG mode inside of the dev.py file. Change to `DEBUG = False`. Also add `ALLOWED_HOSTS = ['*']`. These will be temporary changes to test out the 500 error messaging.
- `npm run build-js`
- Build assets using collectstatic so that scripts and styles will work on site: `./manage.py collectstatic --noinput -v 0`
- Set the feature flag `export FEC_FEATURE_WEBSITE_STATUS=true`. This will turn on the alternative 500 error messaging (see screenshot above for what that looks like).
- Pick a jinja template or a HTML template (our templates all use jinja as the framework even though the extension is .html). 
- Modify that chosen jinja template's extends to reference an invalid file name. Example: `{% extends 'layouts/invalid_name.jinja' %}`
- Run the server `./manage.py runserver`
- Go to the page that uses that modified jinja template. You should see the alternative 500 error message. 
- Test turning off the alternative 500 error message by 
    - Turn off your CMS server
    - Unset the flag: `unset FEC_FEATURE_WEBSITE_STATUS`
- Run the server `./manage.py runserver`
- Refresh the page with the modified jinja template and you should now see the normal/default 500 error message as seen in the screenshot above.

____
